app/command_code_test.go

```diff
@@ -0,0 +1,22 @@
+package app
+
+import (
+	"testing"
+
+	"github.com/mattermost/platform/model"
+)
+
+func TestCodeProviderDoCommand(t *testing.T) {
+	for msg, expected := range map[string]string{
+		"":           "",
+		"foo":        "    foo",
+		"foo\nbar":   "    foo\n    bar",
+		"foo\nbar\n": "    foo\n    bar\n    ",
```

Comment:
indentation is used instead of backticks so that code blocks including the string "```" aren't unintentionally terminated. briefly considered stripping the trailing newline, but decided for code, it's probably best to remain faithful to the user's input, even for whitespace